coordinator/src/services/translation/response/release/release-response.ts

```diff
@@ -1,32 +1,92 @@
 import {fhir, hl7V3} from "@models"
 import * as uuid from "uuid"
+import * as moment from "moment"
 import {toArray} from "../../common"
 import {convertHL7V3DateTimeToIsoDateTimeString} from "../../common/dateTime"
 import {createBundle} from "../../common/response-bundles"
 import {convertResourceToBundleEntry} from "../common"
+import {verifySignature} from "../../../verification/signature-verification"
 
 const SUPPORTED_MESSAGE_TYPE = "PORX_MT122003UK32"
 
-export function createOuterBundle(releaseResponse: hl7V3.PrescriptionReleaseResponse): fhir.Bundle {
+function createInvalidSignatureOutcome(prescription: fhir.Bundle): fhir.OperationOutcome {
+  const extension: fhir.IdentifierReferenceExtension<fhir.Bundle> = {
+    url: "https://fhir.nhs.uk/StructureDefinition/Extension-Spine-supportingInfo-prescription",
+    valueReference: {identifier: prescription.identifier}
+  }
+  return {
+    resourceType: "OperationOutcome",
+    meta: {
+      lastUpdated: moment.utc().format("YYYY-MM-DD[T]HH:mm:ssZ")
+    },
+    extension: [extension],
+    issue: [{
+      severity: "error",
+      code: fhir.IssueCodes.INVALID,
+      details: {
+        coding: [{
+          system: "https://fhir.nhs.uk/CodeSystem/Spine-ErrorOrWarningCode",
+          code: "INVALID",
+          display: "Signature is invalid."
+        }]
+      },
+      expression: ["Provenance.signature.data"]
+    }]
+  }
+}
+
+export function createOuterBundle(releaseResponse: hl7V3.PrescriptionReleaseResponse): fhir.Parameters {
   const releaseRequestId = releaseResponse.inFulfillmentOf.priorDownloadRequestRef.id._attributes.root
   const parentPrescriptions = toArray(releaseResponse.component)
     .filter(component => component.templateId._attributes.extension === SUPPORTED_MESSAGE_TYPE)
```

Comment:
Just spotted this- I wonder if it would be a good idea to create a separate function to do this check and add some comments with a bit of context, e.g. why do we only care about this message type?